src/main/java/edu/utdallas/davisbase/compiler/Compiler.java

```diff
@@ -100,6 +105,9 @@ else if (command instanceof InsertCommandRepresentation) {
               insert.getTable(),
               insert.getValues().get(lcv),
               columnName);
+          if(null!=obj){
+            validateUniqueness(insert.getTable(), columnName, obj);
+          }
           insertObjects.add(new InsertObject(lcv, obj));
```

Comment:
Otherwise, if `obj` _is_ null, we need to validate whether that column is nullable. I'm thinking an analogous `checkIsNullable(tableName, columnName)` method?

Reply:
as part of the validateTypeMatchesSchema the object that is returned already goes through that check to confirm that the object is allowed to be nullable